drivers/interrupt_controller/intc_scr_ipic.c

```diff
@@ -0,0 +1,189 @@
+/*
+ * Copyright (c) 2023 Syntacore
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT scr_ipic
```

Comment:
`scr` is a pretty weird prefix. Use the full `syntacore` and add that to `dts/bindings/vendor-prefixes.txt`

Reply:
Sorry, but I think it's not our task to decide which prefix to use. I will add `scr` to `vendor-prefixes.txt`, and address this proposal to my more experienced colleagues and stakeholders.

Reply:
I checked and Linux is not shipping any vendor prefix for that, so we are free to set the prefix ourselves. So you have to convince me really, really hard that `scr` is a meaningful, complete and self-documenting prefix for `syntacore` and explain why we cannot set that to `syntacore` in the first place.

If you will not be able to do that, I will NACK this until the end of days, even with a positive opinion of your illustrious more experienced colleagues.

Reply:
First of all, I never meant to point on someone's achievements or experience. I just say, that from my point of view, naming decisions are up to product owners from Syntacore side (excuse me, if I'm wrong).

Anyways, on my opinion - if we do `syntacore`, we should use it in every filename, for example: `intc_syntacore_ipic`, which is kind of ambiguous.

Reply:
Why using an abbreviation like `scr` when you could use the full name? What's the advantage? Saving a few characters but sacrificing all the rest?

If one see `scr` in the DT how do they go from there to the actual company name? Guessing?

Have you used the `scr` prefix anywhere else I can take a look?

Reply:
@carlocaione ,

We just paving a ground for our first platform to be upstreamed, so far the only dependency we have from "scr" binding is riscv_machine_timer: https://github.com/zephyrproject-rtos/zephyr/blob/8951bab7e741f3750532d4af68497b9af06a453c/drivers/timer/riscv_machine_timer.c#L66

For some reason we've forgotten to update the bindings at that time. 
As far as I know we have no preference/restrictions over "scr" vs "syntacore" when it comes to DTS bindings. 

But for the vendor-specific function prefixes or vendor-specific CONFIGS we would prefer to use scr_ and _SCR_, since it's our internal code style policy and a lot of code is written with this naming convention. So it will simplify upstreaming / rebasing.

Probably it all would look more consistent if we will use "scr" everywhere, but we'll change vendor-prefixes.txt (and later riscv_machine_timer) if it stylistically better to use "syntacore" for the bindings, no problem

Reply:
> For some reason we've forgotten to update the bindings at that time.

Yes, that should be fixed.

> But for the vendor-specific function prefixes or vendor-specific CONFIGS we would prefer to use scr_ and SCR, since it's our internal code style policy and a lot of code is written with this naming convention. So it will simplify upstreaming / rebasing.

That's fine to me, but the vendor prefix is really a string for the vendor name, no need to shorten that if we can.